h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/remote/SteamExecutorStarter.java

```diff
@@ -112,40 +117,51 @@ private synchronized void sendMessage(Map<String, String> message) throws IOExce
             }
         }
     }
-
+    
     @Override
     public void onMessage(Map<String, String> message) {
         if ("stopXGBoostClusterNotification".equals(message.get(TYPE))) {
             handleStopRequest(message);
         } else {
-            queueResponseMessage(message);
+            queueMessageIfResponse(message);
         }
     }
 
-    private void queueResponseMessage(Map<String, String> message) {
-        LOG.info("Received message response " + message.get(ID));
+    private void queueMessageIfResponse(Map<String, String> message) {
         synchronized (receivedMessages) {
-            receivedMessages.put(message.get(ID), message);
-            receivedMessages.notifyAll();
+            if (receivedMessages.containsKey(message.get(ID))) {
+                LOG.info("Received message response " + message.get(ID));
+                receivedMessages.put(message.get(ID), message);
+                receivedMessages.notifyAll();
+            } else {
+                LOG.debug("Ignoring message " + message.get(ID));
```

Comment:
Should we log the ignored ones at info-level as well ?

Reply:
No since the steam servlet will distribute all messages to all listeners, we do not need to log messages that an listerner ignores